### PR TITLE
[WIP] HTTP SSE consumer example

### PR DESCRIPTION
**Pull request by wvanbergen**
This example HTTP server allows you to consume a partition as an SSE stream. 

### TODO
- [ ] In order to write decent tests for this, I will need a MockClient.

### CODE REVIEW

**Comment by wvanbergen** [comment]
For a potential mock `Client`, I noticed that `Leader` still returns a `*Broker` type, so if it lives in another package we will not be able to use it from sarama itself :(

**Comment by eapache** [comment]
I am coming to the conclusion that we're not going to have any choice but to duplicate our mocks for testing of sarama itself.

File: examples/http_sse/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# SSE server example
+
+This example HTTP server provides a Kafka partition as an SSE stream.
```
Comment (eapache) [inline examples/http_sse/README.md]: what does SSE stand for?

File: examples/http_sse/http_sse.go [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+	if *brokers == "" {
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
+
+	brokerList := strings.Split(*brokers, ",")
+	log.Printf("Kafka brokers: %s", strings.Join(brokerList, ", "))
+
+	config := sarama.NewConfig()
+	client, err := sarama.NewClient(brokerList, config)
```
Comment (eapache) [inline examples/http_sse/http_sse.go]: you can just pass `nil` for all default config

File: examples/http_sse/http_sse.go [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+}
+
+type SSEServer struct {
+	client sarama.Client
+}
+
+func (s *SSEServer) Handler() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		matches := pathMatcher.FindStringSubmatch(r.URL.Path)
+		if len(matches) != 4 {
```
Comment (eapache) [inline examples/http_sse/http_sse.go]: I only see 3 match groups in your regex?

Comment (wvanbergen) [inline examples/http_sse/http_sse.go]: The first entry is the full match, the rest are the captures groups.

File: examples/http_sse/README.md (hunk 1 above)
Comment (wvanbergen) [inline examples/http_sse/README.md]: Server-side events. I should probably add this to the README.

**Comment by eapache** [comment]
Closing as stale.

**Pull request closed without merging**